utils/metadata.py

```diff
@@ -0,0 +1,415 @@
+        dicom_data["AcquisitionDuration"] = None
+
+    if dicom_data["AcquisitionDuration"] is None:
+
+        try:
+            dicom_data["AcquisitionDuration"] = dicom_header[0x0018, 0x9073].value
+        except (AttributeError, KeyError):
+            dicom_data["AcquisitionDuration"] = None
+
+    # Number of dynamics is the number of temporal positions
+    try:
+        dicom_data["NumberOfDynamics"] = dicom_header.NumberOfTemporalPositions
+    except AttributeError:
+        dicom_data["NumberOfDynamics"] = None
+
+    # Calculate field of view
+    try:
+        fov_x = round(dicom_header.PixelSpacing[0] * dicom_header.Columns)
+        fov_y = round(dicom_header.PixelSpacing[1] * dicom_header.Rows)
+        dicom_data["FOV"] = [fov_x, fov_y]
+    except (AttributeError, IndexError):
+        dicom_data["FOV"] = None
+
+    try:
+        dicom_data["SpacingBetweenSlices"] = float(dicom_header.SpacingBetweenSlices)
+    except AttributeError:
+        dicom_data["SpacingBetweenSlices"] = None
+
+    try:
+        dicom_data["PixelSpacing"] = [
+            float(dicom_header.PixelSpacing[0]),
+            float(dicom_header.PixelSpacing[1]),
+        ]
+    except AttributeError:
+        dicom_data["PixelSpacing"] = None
+
+    try:
+        dicom_data["Resolution"] = calculate_resolution(dicom_header)
+    except AttributeError:
+        dicom_data["Resolution"] = None
+
+    try:
+        dicom_data["AcquisitionMatrix"] = dicom_header.AcquisitionMatrix
+    except AttributeError:
+        dicom_data["AcquisitionMatrix"] = None
+
+    try:
+        dicom_data["NumberOfEchos"] = dicom_header[0x2001, 0x1014].value
+    except (AttributeError, KeyError):
+        dicom_data["NumberOfEchos"] = None
+
+    try:
+        dicom_data["NumberOfSlices"] = dicom_header[0x2001, 0x1018].value
+    except (AttributeError, KeyError):
+        dicom_data["NumberOfSlices"] = None
+
+    try:
+        dicom_data["PrepulseDelay"] = dicom_header[0x2001, 0x101B].value
+    except (AttributeError, KeyError):
+        dicom_data["PrepulseDelay"] = None
+
+    try:
+        dicom_data["PrepulseType"] = dicom_header[0x2001, 0x101C].value
+    except (AttributeError, KeyError):
+        dicom_data["PrepulseType"] = None
+
+    try:
+        dicom_data["SliceOrientation"] = dicom_header[0x2001, 0x100B].value
+    except (AttributeError, KeyError):
+        dicom_data["SliceOrientation"] = None
+
+    try:
+        dicom_data["ScanningTechnique"] = dicom_header[0x2001, 0x1020].value
+    except (AttributeError, KeyError):
+        dicom_data["ScanningTechnique"] = None
+
+    try:
+        dicom_data["ScanType"] = dicom_header[0x2005, 0x10A1].value
+    except (AttributeError, KeyError):
+        dicom_data["ScanType"] = None
+
+    return dicom_data
+
+
+def calculate_resolution(dicom_header):
+    """Calculate the voxel resolution from a dicom file header."""
+    # See: https://neurostars.org/t/calculating-voxel-resolution-from-dicom-headers/7091
```

Comment:
I believe this is not valid for most clinical data where the percent sampling impacts the voxel resolution. While I don't disagree that he found an edge-case where this would not suffice as a calculation (i.e., on mosaics), it is necessary for datasets with percent sampling. In addition, the FOV would be skewed for the dataset he presented based. Typically, the FOV is set to columns/rows from the dicom header, and in the example Chris provided, this would be in the thousands, which is what impacts the downstream voxel resolution calculation as I have it.

One option is to just report back the dicom headers that are present and leave it to the user to decide how to calculate their voxel resolution. Otherwise, I would suggest looking at the following references to understand what I am saying by percent sampling - a point that Chris never addressed despite multiple requests to discuss the impact of that parameter on voxel resolution calculations.

- [Report - Example Table 2](https://meddocsonline.org/journal-of-radiology-and-medical-imaging/resolution-snr-signal-averaging-and-scan-time-in-mri-formetastatic-lesion-in-spine-a-case-report.pdf)
- [Chapter - Factors Affecting SNR](http://eknygos.lsmuni.lt/springer/205/29-39.pdf)
- [Forum - Rectangular FOV](http://mriquestions.com/rectangular-fov.html)
- [Forum - FOV](http://mriquestions.com/field-of-view-fov.html)
- [Book - Neuroradiology](https://books.google.com/books?id=86jXU3JrDWsC&pg=PA237&lpg=PA237&dq=percent+phase+fov+impact+on+voxel+resolution&source=bl&ots=m_Fv9nxC1B&sig=ACfU3U19B6jX_Js9HIYIAqMJ72XoP4op9A&hl=en&sa=X&ved=2ahUKEwjKnb-Tv4TqAhV8SjABHY_fDPAQ6AEwEHoECAoQAQ#v=onepage&q=percent%20phase%20fov%20impact%20on%20voxel%20resolution&f=false)

Reply:
As we discussed today, I think we should to stick to the value in the generated nifti metadata for voxel size (pixdim) and compute the FOV from the pixdim and dim values in the nifti header.

Reply:
I will report back DICOM tags as they are exported from the scanner and leave it to the user to decide how to calculate downstream parameters.